param/parameterized.py

```diff
@@ -403,14 +404,15 @@ def iscoroutinefunction(function):
 def _instantiate_param_obj(paramobj, owner=None):
     """Return a Parameter object suitable for instantiation given the class's Parameter object"""
 
-    # Shallow-copy Parameter object
-    p = copy.copy(paramobj)
+    # Shallow-copy Parameter object without the watchers
+    try:
+        watchers = paramobj.watchers
+        paramobj.watchers = {}
+        p = copy.copy(paramobj)
+    finally:
+        paramobj.watchers = watchers
```

Comment:
This obscure code coming back deserves a clear comment explaining why it's done this way...